docs/release_process.md

```diff
@@ -75,224 +71,115 @@ set up and changes are ready to be reviewed.
 
 ## Release to Stage
 
-Every commit to `main` is automatically deployed to the [Stage][stage] server
-via Github Actions and Jenkins.
-
-### Create Release Notes on GitHub
-
+Every commit to `main` is automatically deployed to the [Stage][stage] server via Github Actions and Jenkins.
+
+### PR Merges
+PRs can only be merged when they passes all the required checks:
+* Lint
+* Build
+* Unit Tests
+* E2E Tests
+* Deploy Previews
```

Comment:
Is Deploy Preview actually required? I don't think it matters for the purpose of this doc but I thought I noticed that it wasn't before (either is fine w/ me)

Reply:
It was optional before, but we noticed that the cleanup workflow would otherwise fail and leave a lot of zombie containers running after PRs are closed. So it's probably best to keep it required